kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,866 @@
+                for (IndexType i_gp = 0; i_gp < result_double.size(); ++i_gp) {
+                    const double result = result_double[i_gp];
+                    const double reference = r_entity_database.GetValue(time, 0, i_gp);
+                    if (!CheckValues(result, reference)) {
+                        FailMessage(it_elem->Id(), "Element", result, reference, p_var_double->Name(), -1, i_gp);
+                        check_counter += 1;
+                    }
+                }
+            }
+        }
+        for (auto& p_var_array : mpGPVariableArrayList) {
+            const auto& r_var_database = r_gp_database.GetVariableData(*p_var_array);
+
+            #pragma omp parallel for reduction(+:check_counter) firstprivate(result_array)
+            for (int i = 0; i < static_cast<int>(r_elements_array.size()); ++i) {
+                auto it_elem = it_elem_begin + i;
+
+                const auto& r_entity_database = r_var_database.GetEntityData(i);
+                it_elem->CalculateOnIntegrationPoints(*p_var_array, result_array, r_process_info);
+                for (IndexType i_gp = 0; i_gp < result_array.size(); ++i_gp) {
+                    for (IndexType i_comp = 0; i_comp < 3; ++i_comp) {
+                        const double reference = r_entity_database.GetValue(time, i_comp, i_gp);
+                        if (!CheckValues(result_array[i_gp][i_comp], reference)) {
+                            FailMessage(it_elem->Id(), "Element", result_array[i_gp][i_comp], reference, p_var_array->Name(), i_comp, i_gp);
+                            check_counter += 1;
+                        }
+                    }
+                }
+            }
+        }
+        for (auto& p_var_vector : mpGPVariableVectorList) {
+            const auto& r_var_database = r_gp_database.GetVariableData(*p_var_vector);
+
+            #pragma omp parallel for reduction(+:check_counter) firstprivate(result_vector)
+            for (int i = 0; i < static_cast<int>(r_elements_array.size()); ++i) {
+                auto it_elem = it_elem_begin + i;
+
+                const auto& r_entity_database = r_var_database.GetEntityData(i);
+                it_elem->CalculateOnIntegrationPoints(*p_var_vector, result_vector, r_process_info);
+                for (IndexType i_gp = 0; i_gp < result_vector.size(); ++i_gp) {
+                    for (IndexType i_comp = 0; i_comp < result_vector[i_gp].size(); ++i_comp) {
+                        const double reference = r_entity_database.GetValue(time, i_comp, i_gp);
+                        if (!CheckValues(result_vector[i_gp][i_comp], reference)) {
+                            FailMessage(it_elem->Id(), "Element", result_vector[i_gp][i_comp], reference, p_var_vector->Name(), i_comp, i_gp);
+                            check_counter += 1;
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
These 3 are copy & paste. We can create an auxiliary method and pass the variable container and the check counter.

Reply:
If you look closer ytou will see is not copy paste, there are changes

Reply:
OK. Sorry for the noise. Resolving.